mesa/experimental/jupyter_viz.py

```diff
@@ -59,28 +61,106 @@ def make_model():
+                    space_drawer(model, agent_portrayal)
+            elif layout_type == "Measure":
+                for measure in measures:
+                    if callable(measure):
+                        # Is a custom object
+                        measure(model)
+                    else:
+                        make_plot(model, measure)
+        return main
+
     # 3. Set up UI
-    solara.Markdown(name)
-    UserInputs(user_params, on_change=handle_change_model_params)
-    ModelController(model, play_interval, current_step, set_current_step, reset_counter)
-
-    with solara.GridFixed(columns=2):
-        # 4. Space
-        if space_drawer == "default":
-            # draw with the default implementation
-            make_space(model, agent_portrayal)
-        elif space_drawer:
-            # if specified, draw agent space with an alternate renderer
-            space_drawer(model, agent_portrayal)
-        # otherwise, do nothing (do not draw space)
+
+    with solara.AppBar():
+        solara.AppBarTitle(name)
+
+    grid_layout_initial = [
+        {"h": 12, "i": "0", "moved": False, "w": 5, "x": 0, "y": 0},
+        {"h": 12, "i": "1", "moved": False, "w": 5, "x": 7, "y": 0},
+    ]
+
+    # we need to store the state of the grid_layout ourselves, otherwise it will 'reset'
+    # each time we change resizable or draggable
+    grid_layout, set_grid_layout = solara.use_state(grid_layout_initial)
+
+    # render layout and plot
+
+    # jupyter
+    def render_in_jupyter():
+        with solara.Row():
+            with solara.Card("Controls", margin=1, elevation=2):
+                UserInputs(user_params, on_change=handle_change_model_params)
+                ModelController(
+                    model, play_interval, current_step, set_current_step, reset_counter
+                )
+            with solara.Card("Progress", margin=1, elevation=2):
+                # solara.ProgressLinear(True)
+                solara.Markdown(md_text=f"####Step - {current_step}")
+
+        with solara.Row():
+            # 4. Space
+            if space_drawer == "default":
+                # draw with the default implementation
+                make_space(model, agent_portrayal)
+            elif space_drawer:
+                # if specified, draw agent space with an alternate renderer
+                space_drawer(model, agent_portrayal)
+            # otherwise, do nothing (do not draw space)
 
         # 5. Plots
-        for measure in measures:
-            if callable(measure):
-                # Is a custom object
-                measure(model)
-            else:
-                make_plot(model, measure)
+        with solara.GridFixed(columns=len(measures)):
+            for measure in measures:
+                if callable(measure):
+                    # Is a custom object
+                    measure(model)
+                else:
+                    make_plot(model, measure)
+
+    def render_in_browser():
```

Comment:
Apologies, as I am limited on time and travelling. Could you upload a pic of what the render in browser looks like with say 3 plots. This has been a visual sticking point in the past.

Reply:
I have replicated the measure twice:
<img width="1290" alt="Screenshot 2023-11-02 at 01 28 14" src="https://github.com/projectmesa/mesa/assets/19813545/e2bc3855-073d-4839-8a2d-08790bb43848">

Reply:
Awesome thanks and then with the resize and drag that can be reconfigured anyway correct? (this is super cool)

Reply:
Yes